emcc.py

```diff
@@ -2080,17 +2082,18 @@ def dedup_list(lst):
       if embed_memfile(options):
         shared.Settings.SUPPORT_BASE64_EMBEDDING = 1
 
-      emscripten.run(final, final + '.o.js', shared.replace_or_append_suffix(target, '.mem'))
-      final = final + '.o.js'
+      if wasm_only:
+        final = None
```

Comment:
This seems to change the meaning of `final` in this file. Or was it only for JS before? (maybe related to the test failures?)

Reply:
Yes, final was previously almost always used to hold the current JS filename for each stage of the pipeline.  The one exception was here where it temporarily held the name of the initial wasm file.

With this change we use `final = None` to signify that there is no JS at all.

Reply:
I have a followup that cleans this up more too and avoid the temporary wasm file completely.   See https://github.com/emscripten-core/emscripten/issues/12161 for what we should try to avoid temporary files if possible.

Reply:
Makes sense. How about `final_js` then to indicate that this is the final JS and not anything else?